### ISSUE TITLE
Launch VM from custom image

### ISSUE THREAD

**Issue by colinwebber** [Microsoft/azure-shortcuts-for-java#24]
Firstly, thanks for creating this API - it is a huge improvement over the original.

I have not been able to find a way to launch a VM from a custom image. The standard way of doing this would be to host a publicly accessible template then use resource management as follows:

DeploymentExtended deploymentExtended = ResourceHelper.createTemplateDeploymentFromURI(
					resourceManagementClient,
					resourceGroup,
					DeploymentMode.Incremental,
					deploymentName,
					templateURI,
					templateContentVersion,
					parameters);

However, I cannot find anything similar in the shortcuts implementation - perhaps this hasn't been addressed yet, or perhaps there is a simpler way of achieving this?

My next thought was to extend ..resources.implementation.Subscription in order to implement the above while taking advantage of the rest of the shortcuts API. However resourceManagementClient() has package access - not protected, making this difficult to achieve cleanly.

Kindly advise.

**Comment by martinsawicki** [comment]
RIght, ARM template deployments has just not been modeled yet. I should be able to do that pretty quickly soon, so i'll keep this issue open to track.  Note though the "real" [Azure SDK for Java](https://github.com/Azure/azure-sdk-for-java/) is undergoing a *significant* remake right now (for 1.0) and is being modeled actually after this experimental "shortcuts" project. Although we've still got ways to go, in a few weeks, I expect the new Azure SDK for Java to have at least the same coverage as the "shortcuts". So you should be able to see this sort of simplicity in the 1.0 SDK as well eventually. (I'm now involved in the SDK work as well)
